MapboxNavigation/NavigationMapView.swift

```diff
@@ -902,64 +887,53 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
+            2: 0.5,
+            7: 1])
+        circles.circleRadius = NSExpression(forConstantValue: 10)
+        circles.circleStrokeColor = NSExpression(forConstantValue: UIColor.black)
+        circles.circleStrokeWidth = NSExpression(forConstantValue: 1)
         
         return circles
     }
     
     func routeWaypointSymbolStyleLayer(identifier: String, source: MGLSource) -> MGLStyleLayer {
         let symbol = MGLSymbolStyleLayer(identifier: identifier, source: source)
         
-        symbol.text = MGLStyleValue(rawValue: "{name}")
-        symbol.textFontSize = MGLStyleValue(rawValue: 10)
-        symbol.textHaloWidth = MGLStyleValue(rawValue: 0.25)
-        symbol.textHaloColor = MGLStyleValue(rawValue: .black)
+        symbol.text = NSExpression(forConstantValue: "{name}")
+        symbol.textFontSize = NSExpression(forConstantValue: 10)
+        symbol.textHaloWidth = NSExpression(forConstantValue: 0.25)
+        symbol.textHaloColor = NSExpression(forConstantValue: UIColor.black)
         
         return symbol
     }
     
     func routeStyleLayer(identifier: String, source: MGLSource) -> MGLStyleLayer {
         
         let line = MGLLineStyleLayer(identifier: identifier, source: source)
-        line.lineWidth = MGLStyleValue(interpolationMode: .exponential,
-                                       cameraStops: MBRouteLineWidthByZoomLevel,
-                                       options: [.defaultValue : MGLConstantStyleValue<NSNumber>(rawValue: 1.5)])
-        
-        line.lineColor = MGLStyleValue(interpolationMode: .categorical, sourceStops: [
-            "unknown": MGLStyleValue(rawValue: trafficUnknownColor),
-            "low": MGLStyleValue(rawValue: trafficLowColor),
-            "moderate": MGLStyleValue(rawValue: trafficModerateColor),
-            "heavy": MGLStyleValue(rawValue: trafficHeavyColor),
-            "severe": MGLStyleValue(rawValue: trafficSevereColor)
-            ], attributeName: MBCongestionAttribute, options: [.defaultValue: MGLStyleValue(rawValue: trafficUnknownColor)])
-        
-        line.lineOpacity = MGLStyleValue(interpolationMode: .categorical, sourceStops: [
-            true: MGLStyleValue(rawValue: 1),
-            false: MGLStyleValue(rawValue: 0)
-            ], attributeName: MBCurrentLegAttribute, options: nil)
-        
-        line.lineJoin = MGLStyleValue(rawValue: NSValue(mglLineJoin: .round))
+        line.lineWidth = NSExpression(format: "FUNCTION($zoomLevel, 'mgl_interpolateWithCurveType:parameters:stops:', 'linear', nil, %@)", argumentArray: [MBRouteLineWidthByZoomLevel])
+        line.lineOpacity = NSExpression(forConditional: NSPredicate(format: "\(MBCurrentLegAttribute) == true"), trueExpression: NSExpression(forConstantValue: 1), falseExpression: NSExpression(forConstantValue: 0.85))
+        line.lineColor = NSExpression(format: "%@.severe", [
```

Comment:
@anandthakker @1ec5 this is failing with 

```
Assertion failed: (literal), function operator(), file /Users/distiller/project/include/mbgl/style/conversion/data_driven_property_value.hpp, line 50.
```

Reply:
@bsudekum @1ec5 could we dump the json expression representation that's getting produced by this NSExpression?

Reply:
This expression assumes all traffic is severe. The format should be:

```swift
NSExpression(format: "FUNCTION(%@, 'valueForKey:', congestion)")
```

Alternatively, use an `MGL_MATCH` expression:

```swift
NSExpression(format: "MGL_MATCH(congestion, 'unknown', %@, 'low', %@, 'moderate', %@, …)")
```

Reply:
We can get the JSON expression representation via `NSExpression.mgl_jsonExpressionObject()`. It might be worth printing that out as part of the exception handler. Currently, the exception handler is in code that only knows about mbgl expressions, not NSExpression.

Reply:
This is an expression parsing bug -- reopened here: https://github.com/mapbox/mapbox-gl-native/issues/10849